### DISCUSSION TITLE
Route53 and the IDs returned

### DISCUSSION THREAD

**Discussion by khinkson** [soto-project/soto#737] [General]
I used Soto Route53 yesterday for the first time just to help speed up a dev environment daily setup and teardown. I noticed that certain methods return an ID value but it's not exactly an ID value.

Example
For createHostedZone the response hostedZone.id will look something like "/hostedzone/Z02627942RWLEUJXJILLK". The actual ID is "Z02627942RWLEUJXJILLK". I know this because I cannot take the full path returned and plug it back into other methods that require the zoneID.

Similar for any changes. So the same createHostedZone.changeInfo.id will return something like "/change/C08316092TH67QAEWL9XS". The actual ID is "C08316092TH67QAEWL9XS". And only the actual ID can be used in for example a GetChangeRequest.

And this leads to another issue. In my testing yesterday I (only) once saw an error thrown on the length of the hostedZone.id. It was 33 characters long, one over the expected 32 but this was including the "/hostedzone/" prefix. This makes me wonder if the hostedZone.id length is supposed to include the "/hostedzone/" prefix? I only saw this once in my short testing.

All this to say I found this confusing in the very least. AWS' documentation shows the IDs with the prefix being returned so I that appears to be the expected behaviour. But the 33 character zoneID with prefix surprised me. I googled around and saw the error mentioned once or twice elsewhere which leads me to believe the character limit should be applied to the hostedZone ID without the prefix?

Any thoughts? It's not clear to me from the documentation what the expected behaviour should be. Personally I think the IDs should not be prefixed because we cannot then used them in the other methods that require them. And then there is the validation issue which seems like a bug.

**Comment by adam-fowler** [comment]
Where is this validation being applied?

**Reply by khinkson** [reply]
I haven't been able to re-create the situation since I changed some of the code since. But I suspect it's was an issue with how I understood the ID to work. I likely plugged the ID back into a ChangeResourceRecordSetsRequest without extracting it without the prefix and triggered the validation error.

**Reply by adam-fowler** [reply]
It’s a bit annoying AWS prefix the ids with `/hostedzone/` etc. I can kind of see why they did it. The ids returned are part of the url you would use to interact with a hosted zone. Eg the GetHostedZone operation uses url
```
GET /2013-04-01/hostedzone/Id
```
It seems like this api was defined before many of the SDKs existed.

**Reply by khinkson** [reply]
Yes, I noticed that looking in at the API examples in the AWS documentation. It made me wonder if the /hostedzone/Id is to be used everywhere as the ID instead of switching between the two. 

Anyway, now I know, so I'll keep it in mind.

**Reply by adam-fowler** [reply]
You should pass in just the `id`. The model files we get from AWS add the `/hostedzone/` part to the URL path for you. If you pass in `/hostedzone/{id}` it'll call URL `GET /2013-04-01/hostedzone/hostedzone/{id}` which is obviously wrong

**Comment by khinkson** [comment]
That makes sense. Thanks.
